### PR TITLE
Document OS boot slot changing using the GRUB boot menu

### PR DESCRIPTION
**Pull request by sairon**
## Proposed change

Since home-assistant/supervisor#5276 boot slot will be persisted when user selects it in the GRUB menu. Document this behavior.

This should be merged to the `current` branch once Supervisor 2024.8.1 is released to the stable channel.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation on the GRUB boot loader, detailing how users can select an alternative boot slot during the boot process.
	- Clarified instructions for managing boot slots, improving overall usability for users utilizing GRUB.

### CODE REVIEW

**Comment by c0ffeeca7** [comment]
Drafting this PR as it is waiting for parent

File: source/common-tasks/os.markdown [hunk 1]
```diff
@@ -74,6 +74,8 @@ You can manually define that the previous boot slot is used. This can be useful
     ha os boot-slot other
     ```
 
+Alternatively, if the Operating Systems runs on a platform that uses the GRUB bootloader, a boot menu is presented early in the boot. The alternative boot slot can be selected here, marking it active for future boots if the following boot attempt is successful.
```
Comment (c0ffeeca7) [inline source/common-tasks/os.markdown:77]:
```suggestion
The steps above do not apply if the operating system runs on a platform that uses the GRUB bootloader. In this case, you will be presented with a boot menu early in the boot. 
- Select the alternative boot slot.
- If the following boot attempt is successful, this boot slot is marked as active for future boots.
```
Tiny rephrase suggestion. Feel free to rework if you disagree.

**Comment by sairon** [comment]
@c0ffeeca7 The described functionality is available since Supervisor 2024.9.0 which is [already on stable](https://github.com/home-assistant/version/blob/master/stable.json), can we merge it now?

File: source/common-tasks/os.markdown (hunk 1 above)
Comment (sairon) [reply source/common-tasks/os.markdown:77]: It's not true, the steps still apply, it's just an alternative way how it can be done.

**Review by c0ffeeca7 — APPROVED** [review]

**Pull request merged**